python/graphstorm/config/argument.py

```diff
@@ -203,6 +203,113 @@ def override_arguments(self, cmd_args):
                 setattr(self, f"_{arg_key}", arg_val)
                 print(f"Overriding Argument: {arg_key}")
 
+    def verify_arguments(self, is_train):
+        """ Verify the correctness of arguments.
+
+        Parameters
+        ----------
+        is_train : bool
+            Whether this is for training.
+        """
+        # Trigger the checks in the arguments.
+        _ = self.save_perf_results_path
+        _ = self.profile_path
+        _ = self.graph_name
+        _ = self.backend
+        _ = self.ip_config
+        _ = self.part_config
+        _ = self.node_id_mapping_file
+        _ = self.edge_id_mapping_file
+        _ = self.verbose
+
+        # Data
+        _ = self.node_feat_name
+        _ = self.decoder_edge_feat
+
+        # Evaluation
+        _ = self.eval_fanout
+        _ = self.use_mini_batch_infer
+        _ = self.eval_batch_size
+        _ = self.eval_frequency
+        _ = self.no_validation
```

Comment:
We only need to check these evaluation related arguments when
`if self.no_validation is False`

Reply:
even in the inference script?

Reply:
Inference also accepts no_validation.